### PR TITLE
Bump Roslyn to 2.8.0-beta2-62708-11

### PR DESCRIPTION
**Pull request by KirillOsenkov**
@Therzok there's currently a build failure here:

```
"C:\monodevelop\main\src\core\MonoDevelop.Ide\MonoDevelop.Ide.csproj" (default target) (24) ->
       (CoreCompile target) ->
         MonoDevelop.Ide.TypeSystem\MonoDevelopPersistentStorageLocationService.cs(47,54): error CS0535: 'MonoDevelopPe
       rsistentStorageLocationService' does not implement interface member 'IPersistentStorageLocationService.TryGetSto
       rageLocation(SolutionId)' [C:\monodevelop\main\src\core\MonoDevelop.Ide\MonoDevelop.Ide.csproj]
         MonoDevelop.Ide.TypeSystem\MonoDevelopPersistentStorageLocationService.cs(47,54): error CS0535: 'MonoDevelopPe
       rsistentStorageLocationService' does not implement interface member 'IPersistentStorageLocationService.StorageLo
       cationChanging' [C:\monodevelop\main\src\core\MonoDevelop.Ide\MonoDevelop.Ide.csproj]
```

Could you take over this PR and fix it up? Thanks!

### CODE REVIEW

**Comment by Therzok** [comment]
Still looking at it, getting MEF failures

```
MEF catalog scanning errors encountered.
Microsoft.VisualStudio.Composition.PartDiscoveryException: Failure while scanning type "Microsoft.CodeAnalysis.GenerateEqualsAndGetHashCodeFromMembers.AbstractGenerateEqualsAndGetHashCodeFromMembersCodeRefactoringProvider". ---> System.InvalidOperationException: No importing constructor found.
  at Microsoft.Verify.Operation (System.Boolean condition, System.String message) [0x00003] in <e59308f1b8ae429f87f0bb0a89e1e0d0>:0 
  at Microsoft.VisualStudio.Composition.AttributedPartDiscovery.CreatePart (System.Type partType, System.Boolean typeExplicitlyRequested) [0x0045a] in <1ff596d7134f4c739ec70909207dc9a8>:0 
  at Microsoft.VisualStudio.Composition.PartDiscovery+CombinedPartDiscovery.CreatePart (System.Type partType, System.Boolean typeExplicitlyRequested) [0x00020] in <1ff596d7134f4c739ec70909207dc9a8>:0 
  at Microsoft.VisualStudio.Composition.PartDiscovery+<>c__DisplayClass26_0.<CreateDiscoveryBlockChain>b__0 (System.Type type) [0x00000] in <1ff596d7134f4c739ec70909207dc9a8>:0 
   --- End of inner exception stack trace ---
```

**Comment by Therzok** [comment]
Related roslyn bug: https://github.com/dotnet/roslyn/issues/25427

**Comment by KirillOsenkov** [comment]
Fix to Roslyn available:
https://github.com/dotnet/roslyn/pull/25432

For now I've disabled throwing on VSMEF composition errors because this happens in VS for Windows as well and they survive it. After we get a version of Roslyn with my fix above, I'll reenable checks, perhaps as part of general work of improving our VSMEF integration.

File: main/tests/MonoDevelop.Refactoring.Tests/MonoDevelop.Refactoring.Tests.csproj [hunk 1]
```diff
@@ -133,7 +133,7 @@
       <Name>MonoDevelop.SourceEditor</Name>
       <Private>False</Private>
     </ProjectReference>
-    <ProjectReference Include="..\..\..\external\RefactoringEssentials\RefactoringEssentials\RefactoringEssentials.csproj">
```
Comment (Therzok) [inline main/tests/MonoDevelop.Refactoring.Tests/MonoDevelop.Refactoring.Tests.csproj]: Hah, how?

Comment (KirillOsenkov) [reply main/tests/MonoDevelop.Refactoring.Tests/MonoDevelop.Refactoring.Tests.csproj]: yeah, no idea. Maybe it goes off project Guid.

**Comment by Therzok** [comment]
Do not merge yet, need to verify persistent storage still works after my changes.

**Comment by Therzok** [comment]
C# 7.3 language was added and F# test is failing due to missing rebase.

**Comment by KirillOsenkov** [comment]
Rebased on latest master.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs [hunk 2]
```diff
@@ -128,7 +128,7 @@ internal CompositionManager ()
 
 			var discoveryErrors = catalog.DiscoveredParts.DiscoveryErrors;
 			if (!discoveryErrors.IsEmpty) {
-				throw new ApplicationException ($"MEF catalog scanning errors encountered.\n{string.Join ("\n", discoveryErrors)}");
+				// throw new ApplicationException ($"MEF catalog scanning errors encountered.\n{string.Join ("\n", discoveryErrors)}");
```
Comment (mkrueger) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs]: Can't we add that to our log file  ? Is it important ?

**Comment by KirillOsenkov** [comment]
Added C# language version 7.3.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs (hunk 2 above)
Comment (KirillOsenkov) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs]: This is a good idea. What API can I use to log to the log file?

**Comment by KirillOsenkov** [comment]
Interesting, from binary compat checker report they've introduced a new dependency from MS.CA.EditorFeatures.dll to MS.VS.ImageCatalog.dll. I'll investigate.

https://wrench.internalx.com/Wrench/WebServices/Download.aspx?workfile_id=20018328

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs (hunk 2 above)
Comment (nosami) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs]: MonoDevelop.Core.LoggingService.LogInfo

**Comment by Therzok** [comment]
Persistence is not working. Need to re-review jason's patch to see what we're doing wrong.

**Comment by Therzok** [comment]
Okay, so, removing do-not-merge. Persistent storage now works. But not for Main.sln, since it's not big enough. We seem to have removed a lot of code lately (🎉), and it dropped slightly under the threshold roslyn sets:
Main.sln: 51476919
Threshold: 52428800

**Comment by KirillOsenkov** [comment]
Rebased on latest master and added VSMEF error logging.

If the builds pass, this should be good to go.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs [hunk 3]
```diff
@@ -128,21 +128,24 @@ internal CompositionManager ()
 
 			var discoveryErrors = catalog.DiscoveredParts.DiscoveryErrors;
 			if (!discoveryErrors.IsEmpty) {
-				// throw new ApplicationException ($"MEF catalog scanning errors encountered.\n{string.Join ("\n", discoveryErrors)}");
+				var message = $"MEF catalog discovery errors encountered.{Environment.NewLine}{string.Join (Environment.NewLine, discoveryErrors)}";
+				LoggingService.LogInfo (message);
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs]: I'd rather log every exception separately:
```
foreach (var error in discoveryErrors) {
    LoggingService.LogInfo ("MEF catalog discovery error encountered", error);
}
```

Same below with composition errors.

Comment (KirillOsenkov) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs]: sure

Comment (KirillOsenkov) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Composition/CompositionManager.cs]: Fixed

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Highlighting/EditorThemeColors.cs [hunk 4]
```diff
@@ -139,22 +139,45 @@ public static class EditorThemeColors
 		public static readonly string UserTypesDelegates = "entity.name.delegate";
 		public static readonly string UserTypesMutable = "entity.name.mutable";
 
+		public static readonly string UserField = "entity.name.field";
+		[Obsolete ("Use UserField")]
```
Comment (mhutch) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Editor.Highlighting/EditorThemeColors.cs]: Did we really need to obsolete all these?

**Pull request merged**